### ISSUE TITLE
[2.3beta1] Wrong permissions für {bbb-html5,freeswitch,bbb-web,bbb_etherpad} files (security issue)

### ISSUE THREAD

**Issue by schrd** [bigbluebutton/bigbluebutton#11752]
**Describe the bug**
Unit files package of the packages bbb-html5,freeswitch,bbb-web,bbb_etherpad have wrong owner. All systemd unit files must be owned by root. The following output is for bbb-html5, but its the same for the other packages.

~~~
dpkg -c /var/cache/apt/archives/bbb-html5_1%3a2.3.0-1511_amd64.deb |grep service
-rw-r--r-- meteor/meteor      476 2021-03-24 20:48 ./usr/share/meteor/bundle/programs/server/npm/node_modules/@icons/material/svg/room-service.svg
-rw-r--r-- meteor/meteor     1056 2021-03-24 20:48 ./usr/share/meteor/bundle/programs/server/npm/node_modules/caniuse-lite/data/features/serviceworkers.js
-rw-r--r-- meteor/meteor      564 2021-03-24 20:48 ./usr/lib/systemd/system/bbb-html5-backend@.service
-rw-r--r-- meteor/meteor      293 2021-03-24 20:48 ./usr/lib/systemd/system/disable-transparent-huge-pages.service
-rw-r--r-- meteor/meteor      534 2021-03-24 20:48 ./usr/lib/systemd/system/bbb-html5.service
-rw-r--r-- meteor/meteor      553 2021-03-24 20:48 ./usr/lib/systemd/system/bbb-html5-frontend@.service
-rw-r--r-- meteor/meteor      897 2021-03-24 20:40 ./etc/systemd/system/mongod.service
~~~

This means that an attacker who is able to break bbb-html5 can be root on the machine at the next reboot.

The situation is even worse: all files in the package are owned by `meteor/meteor`. No file in any package except for a config files should be owned by any other user than root.

~~~
[root@bbb-node /etc/bigbluebutton]# dpkg -c /var/cache/apt/archives/bbb-html5_1%3a2.3.0-1511_amd64.deb |grep meteor/meteor|wc -l
40317
[root@bbb-node /etc/bigbluebutton]# dpkg -c /var/cache/apt/archives/bbb-html5_1%3a2.3.0-1511_amd64.deb |wc -l
40317
~~~

**Installation type**
manual. Just installed the packages. This is a packaging bug and should be fixed in packages, not by quirking bbb-install.

**Comment by danimo** [comment]
Note: This is a regression over 2.2 (even though new files were added), see #10831.

**Comment by ffdixon** [comment]
Thanks for reporting this -- it does look like a regression.  We'll focus on fixing this soon.

**Comment by ffdixon** [comment]
We have the permissions fixed for `bbb-html5` package

```
root@droplet-5664:/usr/lib/systemd/system# ls -al
total 64
drwxr-xr-x  2 root root 4096 Mar 25 23:15 .
drwxr-xr-x 12 root root 4096 Mar 25 22:44 ..
-rw-r--r--  1 root root  564 Mar 25 23:12 bbb-html5-backend@.service
-rw-r--r--  1 root root  553 Mar 25 23:12 bbb-html5-frontend@.service
-rw-r--r--  1 root root  534 Mar 25 23:12 bbb-html5.service
-rw-r--r--  1 root root  262 Mar 25 19:10 bbb-rap-caption-inbox.service
-rw-r--r--  1 root root  460 Mar 25 19:10 bbb-rap-resque-worker.service
-rw-r--r--  1 root root  307 Mar 25 19:10 bbb-rap-starter.service
-rw-r--r--  1 root root  412 Mar 18 16:15 bbb-webrtc-sfu.service
-rw-r--r--  1 root root   56 Mar 25 19:10 bbb_record_core.slice
-rw-r--r--  1 root root  293 Mar 25 23:12 disable-transparent-huge-pages.service
-rw-r--r--  1 root root  448 Mar 22 22:41 etherpad.service
-rw-r--r--  1 root root  725 Mar 25 23:15 kurento-media-server-8888.service
-rw-r--r--  1 root root  725 Mar 25 23:15 kurento-media-server-8889.service
-rw-r--r--  1 root root  725 Mar 25 23:15 kurento-media-server-8890.service
-rw-r--r--  1 root root  503 Mar 18 16:15 kurento-media-server.service
```

and

```
# dpkg -c bbb-html5_1%3a2.3.0-1521_amd64.deb | grep service
-rw-r--r-- root/root       476 2021-03-25 23:12 ./usr/share/meteor/bundle/programs/server/npm/node_modules/@icons/material/svg/room-service.svg
-rw-r--r-- root/root      1056 2021-03-25 23:12 ./usr/share/meteor/bundle/programs/server/npm/node_modules/caniuse-lite/data/features/serviceworkers.js
-rw-r--r-- root/root       564 2021-03-25 23:12 ./usr/lib/systemd/system/bbb-html5-backend@.service
-rw-r--r-- root/root       293 2021-03-25 23:12 ./usr/lib/systemd/system/disable-transparent-huge-pages.service
-rw-r--r-- root/root       534 2021-03-25 23:12 ./usr/lib/systemd/system/bbb-html5.service
-rw-r--r-- root/root       553 2021-03-25 23:12 ./usr/lib/systemd/system/bbb-html5-frontend@.service
-rw-r--r-- root/root       897 2021-03-25 23:04 ./etc/systemd/system/mongod.service
```

This will be in the next beta release for 2.3. Looking into the other packages.

**Comment by blueiceprj** [comment]
I have same problem. Is there any solution?

**Comment by paulmenzel** [comment]
Three files are left in BBB 2.3.3:

```
$ more /etc/bigbluebutton/bigbluebutton-release
BIGBLUEBUTTON_RELEASE=2.3.3
$ ls -al /lib/systemd/system | grep -v root
total 824
-rw-r--r--  1 bigbluebutton bigbluebutton   527 Jun  3 22:14 bbb-web.service
-rw-r--r--  1 freeswitch    daemon           54 Jun  3 02:50 freeswitch.conf
-rw-r--r--  1 freeswitch    daemon         1461 Jun  3 02:50 freeswitch.service
```

These are in the packages *bbb-web* and *bbb-freeswitch-core*.

**Comment by paulmenzel** [comment]
Sorry, two files. The configuration file can be ignored I guess.

**Comment by ffdixon** [comment]
Thanks Paul for pointing these out -- will fix in a coming update to 2.3.

**Comment by antobinary** [comment]
Retagging to 2.5 where the build/ directory contains all the packaging related files (open source). Once resolved we can backport to all maintained versions

**Comment by antobinary** [comment]
Resolved in #14110 on 2.4, the fix will be merged into 'develop' for 2.5 shortly. Thanks for the fixes @schrd !!
